apps/services/form-system/src/app/modules/forms/forms.service.ts

```diff
@@ -0,0 +1,359 @@
+        isCommon: listType.isCommon,
+      } as ListTypeDto)
+    })
+
+    return listTypesDto
+  }
+
+  private setArrays(form: Form): FormDto {
+    const formDto: FormDto = {
+      id: form.id,
+      organizationId: form.organizationId,
+      name: form.name,
+      urlName: form.urlName,
+      invalidationDate: form.invalidationDate,
+      created: form.created,
+      modified: form.modified,
+      isTranslated: form.isTranslated,
+      applicationDaysToRemove: form.applicationDaysToRemove,
+      derivedFrom: form.derivedFrom,
+      stopProgressOnValidatingScreen: form.stopProgressOnValidatingScreen,
+      completedMessage: form.completedMessage,
+      testimonyTypes: [],
+      applicants: [],
+      sections: [],
+      screens: [],
+      fields: [],
+    }
+
+    form.testimonyTypes?.map((testimonyType) => {
+      formDto.testimonyTypes?.push({
+        id: testimonyType.id,
+        name: testimonyType.name,
+        description: testimonyType.description,
+        type: testimonyType.type,
+      } as FormTestimonyTypeDto)
+    })
+
+    form.applicants?.map((applicant) => {
+      formDto.applicants?.push({
+        id: applicant.id,
+        applicantType: applicant.applicantType,
+        name: applicant.name,
+      } as FormApplicantDto)
+    })
+
+    form.sections.map((section) => {
+      formDto.sections?.push({
+        id: section.id,
+        name: section.name,
+        created: section.created,
+        modified: section.modified,
+        sectionType: section.sectionType,
+        displayOrder: section.displayOrder,
+        waitingText: section.waitingText,
+        isHidden: section.isHidden,
+        isCompleted: section.isCompleted,
+        callRuleset: section.callRuleset,
+      } as SectionDto)
+      section.screens?.map((screen) => {
+        formDto.screens?.push({
+          id: screen.id,
+          sectionId: section.id,
+          name: screen.name,
+          created: screen.created,
+          modified: screen.modified,
+          displayOrder: screen.displayOrder,
+          isHidden: screen.isHidden,
+          multiset: screen.multiset,
+        } as ScreenDto)
+        screen.fields?.map((field) => {
+          formDto.fields?.push({
+            id: field.id,
+            screenId: screen.id,
+            name: field.name,
+            created: field.created,
+            modified: field.modified,
+            displayOrder: field.displayOrder,
+            description: field.description,
+            isHidden: field.isHidden,
+            isPartOfMultiset: field.isPartOfMultiset,
+            fieldSettings:
+              this.fieldSettingsMapper.mapFieldTypeToFieldSettingsDto(
+                field.fieldSettings,
+                field.fieldType,
+              ),
+            fieldType: field.fieldType,
+          } as FieldDto)
+        })
+      })
+    })
+
+    return formDto
+  }
+
+  private async createFormTemplate(form: Form): Promise<void> {
```

Comment:
Could also use:

```javascript
await this.secionModel.bulkCreate([
  // list of sections
])
```

Reply:
Three of the four sections fit into bulkcreate, but I need the id of the fourth one as foreign key for the screen